[BUU] Add new product images

dacook opened the pull request:
#### What? Why?

- Closes #11065 

Re-using the edit image form, because they're basically the same.

#### What should we test?
We can now test all scenarios as in the above issue, for both adding a new image, and updating.


#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [x] Feature toggled

<!-- Choose a pull request title above which explains your change to a
     a user of the Open Food Network app. -->

The title of the pull request will be included in the release notes.

mkllnk commented on spec/system/admin/products_v3/products_spec.rb:
```diff
@@ -354,231 +324,306 @@
-          page.find(".vertical-ellipsis-menu").click
-          click_link('Clone')
+          # Upload a new image file
+          attach_file 'image[attachment]', Rails.public_path.join('500.jpg'), visible: false
```
I think that it would be better style use a standard spec helper:
```suggestion
          attach_file 'image[attachment]', file_fixture("logo.png"), visible: false
```

rioug reviewed (approved):
Looks good :+1:

RachL commented:
@dacook which error messages have you implemented so far? I tried to upload a wrong image file for example and faced an error 500, so I guess we are not handling those yet 😁 

Also on staging the modal appears with a lot of empty space on the right:

![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/6525576/55b0483f-1872-4fc1-a132-8aefd6951cb5)

Which is a bit different than on the prototype
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/6525576/441e4502-71fb-4f5c-9433-3c2ac14e30f0)

Is this something easy to fix?

Finally FYI the "edit" button is jumping on the top of the image when the modal is open. I don't think we need to spend time on this though, just a fun fact :) 

![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/6525576/c4938572-6995-4a79-893c-89b84e2b7bed)

Otherwise it seems to work fine 💪

dacook commented on spec/system/admin/products_v3/products_spec.rb:
```diff
@@ -324,6 +324,38 @@
+        end
+
+        expect(page).to have_content "Loading"
+        expect(page).to have_content "Image has been successfully updated"
```
You're right, and your prediction came true, hence I removed it in a later commit ;)

dacook commented:
Thansk @RachL , there shouldn't have been a 500 error, I will take a look.
I'll also try and fix those style issues if I can.

dacook commented:
@mkllnk I've added the last three commits for re-review.

The error was logged in bugsnag for [open-food-france](https://app.bugsnag.com/open-food-france/coopcircuits/errors/6587d8892d5cf5000963ceaf?filters[event.unhandled]=true&filters[event.since]=7d&filters[app.release_stage]=staging&sort=events).
To save you having to log in, here's the relevant code from the stacktrace in case you're interested:
`gems/activestorage-7.0.8/lib/active_storage/downloader.rb:39`
```
      def verify_integrity_of(file, checksum:)
        unless OpenSSL::Digest::MD5.file(file).base64digest == checksum
          raise ActiveStorage::IntegrityError
        end
      end
  end
```

mkllnk reviewed (approved):
Excellent new spec.

dacook commented:
Thanks, removed last commit and rebased.

@RachL  I've addressed the 500 error that you received, can you please test again?
Regarding the styling issues, I'll address these in a separate PR as it requires more general style fixes.

RachL commented:
@dacook I still have a snail on the new page and a clean error message on the old page:

![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/6525576/f2e0fbc3-a3fc-4191-a2cb-308bea5c00ce)

but maybe there was a problem in staging?

RachL commented:
Ok weird indeed it's working on staging UK!
